File: go/net_http/send_contact.go
// Send Contact Message - إرسال جهة اتصال - رابطہ بھیجیں
// Uses net/http only - استخدام net/http فقط - صرف net/http استعمال
// Note: Contact uses path "message/contact" - different from the 7 existing types
package main

import (
	"bytes"
	"encoding/base64"
	"encoding/json"
	"fmt"
	"io"
	"net/http"
)

// Config - الإعدادات - ترتیبات
const (
	appKey    = "YOUR_APP_KEY"
	apiSecret = "YOUR_API_SECRET"
	projectID = "YOUR_PROJECT_ID"
	recipient = "966500000000"
)

func main() {
	// Request body - جسم الطلب - درخواست کا جسم
	// هيكل مختلف: path message/contact مع params مباشرة
	// Different structure: path message/contact with direct params
	payload := map[string]interface{}{
		"path": "message/contact",
		"params": map[string]interface{}{
			"phone": recipient,
			"contacts": []map[string]interface{}{
				{
					"name": map[string]string{
						"formatted_name": "Ahmed Ali",
						"first_name":     "Ahmed",
						"last_name":      "Ali",
					},
					"phones": []map[string]interface{}{
						{"phone": "+966501234567", "type": "CELL"},
					},
				},
			},
		},
	}

	body, err := json.Marshal(payload)
	if err != nil {
		fmt.Printf("JSON marshal error: %v\n", err)
		return
	}

	url := fmt.Sprintf("https://api-users.4jawaly.com/api/v1/whatsapp/%s", projectID)
	req, err := http.NewRequest("POST", url, bytes.NewReader(body))
	if err != nil {
		fmt.Printf("Request creation error: %v\n", err)
		return
	}

	auth := base64.StdEncoding.EncodeToString([]byte(appKey + ":" + apiSecret))
	req.Header.Set("Authorization", "Basic "+auth)
	req.Header.Set("Content-Type", "application/json")

	client := &http.Client{}
	resp, err := client.Do(req)
	if err != nil {
		fmt.Printf("Request failed: %v\n", err)
		return
	}
	defer resp.Body.Close()

	respBody, _ := io.ReadAll(resp.Body)
	fmt.Printf("Status: %d\n", resp.StatusCode)
	fmt.Printf("Response: %s\n", string(respBody))
}

File: php/legacy/send_contact.php
<?php
/**
 * إرسال جهة اتصال عبر واتساب
 * Send contact via WhatsApp
 * واٹس ایپ کے ذریعے رابطہ بھیجیں
 */

// إعدادات الاتصال - Connection settings - رابطہ کی ترتیبات
$app_key = "your_app_key";
$api_secret = "your_api_secret";
$project_id = "your_project_id";
$recipient = "966500000000";

// بيانات جهة الاتصال - Contact data - رابطہ کا ڈیٹا
$contacts = array(
    array(
        "name" => array(
            "formatted_name" => "Ahmed Ali",
            "first_name" => "Ahmed",
            "last_name" => "Ali"
        ),
        "phones" => array(
            array(
                "phone" => "+966501234567",
                "type" => "CELL"
            )
        )
    )
);

// بناء الطلب - Build request - درخواست بنائیں
// ملاحظة: جهة الاتصال تستخدم هيكل طلب مختلف عن الأنواع الأخرى
// Note: Contact uses a different request structure than other types
$base_url = "https://api-users.4jawaly.com/api/v1/whatsapp/" . $project_id;
$auth = base64_encode($app_key . ":" . $api_secret);

$payload = array(
    "path" => "message/contact",
    "params" => array(
        "phone" => $recipient,
        "contacts" => $contacts
    )
);

// إرسال الطلب عبر cURL - Send request via cURL - cURL کے ذریعے درخواست بھیجیں
$ch = curl_init($base_url);
curl_setopt($ch, CURLOPT_POST, true);
curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($payload));
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_HTTPHEADER, array(
    "Content-Type: application/json",
    "Authorization: Basic " . $auth
));

$response = curl_exec($ch);
$http_code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
$error = curl_error($ch);
curl_close($ch);

// معالجة النتيجة - Handle result - نتیجہ سنبھالیں
if ($error) {
    echo "خطأ في الاتصال: " . $error . "\n";
    echo "Connection error: " . $error . "\n";
    echo "رابطہ میں خرابی: " . $error . "\n";
    exit(1);
}

echo "رمز الاستجابة: " . $http_code . "\n";
echo "Response code: " . $http_code . "\n";
echo "جوابی کوڈ: " . $http_code . "\n";
echo "\nالاستجابة الكاملة:\nFull response:\nمکمل جواب:\n";
echo $response . "\n";
